### DISCUSSION TITLE
general-part-first contradicts hint-at-types examples

### DISCUSSION THREAD

**Discussion by floprr** [airbnb/swift#187] [General]
Rule **Names should be written with their most general part first and their most specific part last** is not usual because ex. `infoLabel`, `cancelButton` are very common names in iOS which is also encouraged in the next rule **Include a hint about type in a name if it would otherwise be ambiguous** examples.

https://github.com/airbnb/swift#general-part-first
https://github.com/airbnb/swift#hint-at-types

**Comment by garrettlarson** [comment]
I don't think these rules contradict each other. We recommend adding a type hint if the name is ambiguous without one, which implies the type hint is the most specific part and should go at the end. In your example, it distinguishes `cancelButton` from alternatives like `cancelText` or `cancelDialog`.

That said, these rules are pretty vague as written ("depends on context" and "if it would be ambiguous"). Can you think of a way to phrase them more clearly?
